### PR TITLE
FFI: Expose RoomSendQueue::unwedge to allow resending.

### PR DESCRIPTION
**Pull request by pixlwave**
This PR adds the `_and_resend` part of #3909 as it's own method so a user can re-verify on a different device that supports it, and then come back to the device using the FFI and send the message that failed to be sent.

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 1]
```diff
@@ -810,6 +810,23 @@ impl Room {
...
+    /// the user may have re-verified on a different device and would now
+    /// like to send the failed message that's waiting on this device.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_id` - The send queue transaction identifier of the local
+    ///  echo the send error applies to.
+    pub async fn resend(&self, transaction_id: String) -> Result<(), ClientError> {
+        let transaction_id: OwnedTransactionId = transaction_id.into();
+        self.inner.send_queue().unwedge(&transaction_id).await?;
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room.rs:827]:
```suggestion
        let transaction_id: &TransactionId = transaction_id.into();
        self.inner.send_queue().unwedge(transaction_id).await?;
```

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 2]
```diff
@@ -810,6 +810,23 @@ impl Room {
 
         Ok(())
     }
+
+    /// Attempt to manually resend messages that failed to send due to issues
+    /// that should now have been fixed. This is useful for example, when
+    /// there's a
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room.rs:816]: nit: let's add some breathing space:

```suggestion
    /// that should now have been fixed.
    ///
    /// This is useful for example, when there's a
```

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 3]
```diff
@@ -810,6 +810,23 @@ impl Room {
...
+    /// `SessionRecipientCollectionError::VerifiedUserChangedIdentity` error;
+    /// the user may have re-verified on a different device and would now
+    /// like to send the failed message that's waiting on this device.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_id` - The send queue transaction identifier of the local
+    ///  echo the send error applies to.
+    pub async fn resend(&self, transaction_id: String) -> Result<(), ClientError> {
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room.rs:825]: Could we call this `try_resend()` instead?

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 4]
```diff
@@ -810,6 +810,23 @@ impl Room {
...
+    /// that should now have been fixed. This is useful for example, when
+    /// there's a
+    /// `SessionRecipientCollectionError::VerifiedUserChangedIdentity` error;
+    /// the user may have re-verified on a different device and would now
+    /// like to send the failed message that's waiting on this device.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_id` - The send queue transaction identifier of the local
+    ///  echo the send error applies to.
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room.rs:824]: Bad copy pasto, nowhere else we do mention a "send error".

File: bindings/matrix-sdk-ffi/src/room.rs (hunk 1 above)
Comment (pixlwave) [reply bindings/matrix-sdk-ffi/src/room.rs:827]: Ah nice, didn't know you could `into` a borrowed type.

File: bindings/matrix-sdk-ffi/src/room.rs (hunk 3 above)
Comment (pixlwave) [reply bindings/matrix-sdk-ffi/src/room.rs:825]: Sure :)

File: bindings/matrix-sdk-ffi/src/room.rs (hunk 4 above)
Comment (pixlwave) [reply bindings/matrix-sdk-ffi/src/room.rs:824]: 🍝

File: bindings/matrix-sdk-ffi/src/room.rs (hunk 1 above)
Comment (pixlwave) [reply bindings/matrix-sdk-ffi/src/room.rs:827]: I had to add an `as_str()` to this to make the compiler happy.

**Review by bnjbvr — APPROVED** [review]

**Pull request merged**